### ISSUE TITLE
Making double-iframe destination url dynamic

### ISSUE THREAD

**Issue by 1000TurquoisePogs** [zowe/zlux#558]
The current problem with double-iframe apps is that it's not a part of the framework, it's just a trick we have applied, so the framework has no ability to do substitutions of the URL and therefore it must be hardcoded.
Hardcoding becomes a problem when multiple routes to the same server are present. Different IPs that are visible to some users but not visible to others leads to the hardcoding resulting in some apps that cant be seen by some users.

The solution is to incorporate double-iframe logic such that the framework can make substitutions based on configuration, especially instance environment variables.

Today the double iframe html file is a static file that is served by the static file handling part of the app-server. I don't think the static file handling code should be changed, for performance reasons.

**Instead, we should make a new endpoint on the app-server,**
**/ZLUX/plugins/org.abc.my.iframe/iframe/**
This is as opposed to the current single-or-double iframe url
**/ZLUX/plugins/org.abc.my.iframe/web/index.html**

.../iframe will be a REST api, that when called will look at the pluginDefinition of the plugin to see if there is an attribute of the webContent section that describes the destination of the iframe, but with substitutions allowed.
Consider:
```
webContent: {
  framework: iframe,
  destination: "https://${ZOWE_EXPLORER_HOST}:${MY_PORT}/index.html"
}
```

the keyword `destination` as opposed to `startingPage` will be the difference in this enhancement to the plugin definition.
The server can use ALL/ANY of its environment variable knowledge from zowe to substitute the `${}` content it sees in the destination string. This allows for even destinations that are not just ZOWE_EXPLORER_HOST.

well, if contacting .../iframe, the return value can be a semi-hardcoded html:
```
<html>
    <body>
        <iframe 
            id="zluxIframe"
            src="https://mymainframe:29601/index.html" 
            style="position:fixed; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999999;">
            Your browser doesn't support iframes
        </iframe>
    </body>
</html>
```

ultimately, the `src` attribute is the dynamic part, with the rest being hardcoded, yet, from the perspective of the browser it appears to be a normal webpage, so it should be rendered correctly.

**Comment by 1000TurquoisePogs** [comment]
an example
before, jes explorer has the startingPage attribute, which goes here to get the iframe html as seen in the previous comment
GET https://<host>:8544/ZLUX/plugins/org.zowe.explorer-jes/web/index.html
after, jes explorer will have the destination attribute, which implies the following url, which will still return html, but the src attribute will be dynamically determined by the route code in the app server
GET https://<host>:8544/ZLUX/plugins/org.zowe.explorer-jes/iframe/

**Comment by NakulManchanda** [comment]
Thanks @1000TurquoisePogs 
it will be a great enhancement for external iframe apps,
- Need Doc changes
- Zowe-install-packaging changes for explorers and apicatalog,

i believe its a **non-breaking change**, as we add new property instead of changing existing setup in any way.
`GET https://:8544/ZLUX/plugins/org.zowe.explorer-jes/iframe/`

**Comment by NakulManchanda** [comment]
Will incorporate same changes in `sample-trial-app` as well, for demonstration

**Comment by NakulManchanda** [comment]
It will potentially fix issue the issue for ZD&T machines, where ZOWE_EXPLORER_HOST is internal & machine is access via External IP on client side.

![image](https://user-images.githubusercontent.com/4791635/97046436-64066700-1545-11eb-8988-cf5da3822e44.png)

**Comment by NakulManchanda** [comment]
ZLUX IFrame install script changes:
https://github.com/zowe/zowe-install-packaging/pull/1801/files

Explorer JES configure changes:
https://github.com/zowe/explorer-jes/pull/236/files

Virtual Desktop Changes:
https://github.com/zowe/zlux-app-manager/pull/296/files

ZLUX Server Framework changes:
https://github.com/zowe/zlux-server-framework/pull/257/files

**Comment by 1000TurquoisePogs** [comment]
solved in zowe 1.19.0
